client/src/pages/Notification/NotificationContext.tsx

```diff
@@ -0,0 +1,26 @@
+import { 
+  NotificationButtonsProps
+ } from 'pages/Notification/NotificationButtons/NotificationButtons';
+import { createContext, useContext } from 'react';
+
+export type NotificationsChannel = 'SLACK' | 'DISCORD' | 'EMAIL';
+
+export type NotificationContextProps = {
+  [key in NotificationsChannel]: NotificationButtonsProps;
+};
+
+const MockedNotificationButtonsProps: NotificationButtonsProps = {
+  onUpdateClick: () => {},
+  onTestClick: () => {},
+  onDeleteClick: () => {},
+};
+
+export const NotificationContext = createContext<NotificationContextProps>({
+  SLACK: MockedNotificationButtonsProps,
+  DISCORD: MockedNotificationButtonsProps,
+  EMAIL: MockedNotificationButtonsProps,
+});
+
+export const useNotification = () => {
+  return useContext(NotificationContext);
+};
```

Comment:
Consider moving logic of updating, testing and removing channels to button themselves. 
Add to context fields such as NotificationChannel to indicate which channel is open, and something like "Row" to use it's date to populate update modal.  You would also need to move modal state to the context provider parent